terra_accounts/models.py

```diff
@@ -48,13 +49,13 @@ def get_by_natural_key(self, username):
 
     def get_all_terra_permissions(self):
         if self.is_active:
-            if self.is_superuser:
-                perms = TerraPermission.objects.all()
-            else:
-                perms = self.user_permissions.filter(terrapermission__isnull=False)
+            perms = TerraPermission.objects.all()
+
+            if not self.is_superuser:
+                perms = perms.filter(Q(pk__in=self.user_permissions.all()) |
+                                     Q(group__in=self.groups.all()))
```

Comment:
Why don't use a mechanism like standard auth backend but filtering on terrapermissions?
https://github.com/django/django/blob/52936efacec4df05df2d9872c09c3332335bf21b/django/contrib/auth/backends.py#L21

Reply:
This required to customize and change auth backends in all projects using these app.